src/nodeinfo.hpp

```diff
@@ -57,20 +61,26 @@ struct NodeInfo {
     std::unordered_map<std::string, std::string> outputNameAliases;
     std::optional<size_t> demultiplyCount;
     std::optional<std::string> gatherFromNode;
+    NodeLibrary nodeLibrary;
+    parameters_t parameters;
 
     NodeInfo(NodeKind kind,
         const std::string& nodeName,
         const std::string& modelName = "",
         std::optional<model_version_t> modelVersion = std::nullopt,
         std::unordered_map<std::string, std::string> outputNameAliases = {},
         std::optional<size_t> demultiplyCount = std::nullopt,
-        std::optional<std::string> gatherFromNode = std::nullopt) :
+        std::optional<std::string> gatherFromNode = std::nullopt,
+        const NodeLibrary& nodeLibrary = {},
```

Comment:
I think the optional<NodeLibrary> would be more readable -> it would be clearer that this is optional

Probably we should pack this something like:
std::variant<CustomNodeOptions, DLNodeOptions>

Reply:
We agreed to resign from std::variant as it would include to big refactor for now.